### ISSUE TITLE
[Bug] ShowHiddenThings ModView

### ISSUE THREAD

**Issue by ColynOrg** [Vendicated/Vencord#2708]
### Discord Account

CJStylesOrg

### What happens when the bug or crash occurs?

the roles will not show up on the mod view at all despite the user having a role shown in their profile. i have tested this on Bread (304260051915374603) whom is a plugin developer in BetterDiscord. cases like this are rare but i managed to stumble upon this.
![Preview](https://github.com/user-attachments/assets/3bdc984a-45f3-4607-8971-8d58e1138036)

### What is the expected behaviour?

i expect the plugin to retrieve the users role and display them in the mod view, im unaware if this is related to discord or the plugin itself.

### How do you recreate this bug or crash?

1. press the mod view button on peoples profile until you notice the roles don't show up on the mod view despite them actually having roles.

### Errors

```
GET https://canary.discord.com/api/v9/guilds/86004744966914048/audit-logs?limit=50&target_id=304260051915374603 403 (Forbidden)
A._end @ web.5654e7cdcca52e5b08a1.js:12
A.end @ web.5654e7cdcca52e5b08a1.js:12
(anonymous) @ web.5654e7cdcca52e5b08a1.js:12
l.then @ web.5654e7cdcca52e5b08a1.js:12
E @ web.5654e7cdcca52e5b08a1.js:12
(anonymous) @ web.5654e7cdcca52e5b08a1.js:12
I @ web.5654e7cdcca52e5b08a1.js:12
t @ e70441093a3116eaa948.js:1
a @ e70441093a3116eaa948.js:1
S @ e70441093a3116eaa948.js:1
e @ c7f4b1b26af37e6d391f.js:1
(anonymous) @ c7f4b1b26af37e6d391f.js:1
sF @ web.5654e7cdcca52e5b08a1.js:12
oV @ web.5654e7cdcca52e5b08a1.js:12
oD @ web.5654e7cdcca52e5b08a1.js:12
r5 @ web.5654e7cdcca52e5b08a1.js:12
(anonymous) @ web.5654e7cdcca52e5b08a1.js:12
oF @ web.5654e7cdcca52e5b08a1.js:12
oO @ web.5654e7cdcca52e5b08a1.js:12
N @ web.5654e7cdcca52e5b08a1.js:12
L @ web.5654e7cdcca52e5b08a1.js:12
T.C.<computed> @ web.5654e7cdcca52e5b08a1.js:1
D @ web.5654e7cdcca52e5b08a1.js:1
b @ web.5654e7cdcca52e5b08a1.js:1
```

### Request Agreement

- [X] I am using Discord Stable or tried on Stable and this bug happens there as well
- [X] I have read the requirements for opening an issue above

**Comment by Sqaaakoi** [comment]
discord doesn't show the highest role there by design 
![Mod view roles component, with `filter(e=>null != e && e !== t.highestRoleId)` selected](https://github.com/user-attachments/assets/48a9366d-780d-4d4b-8886-f3c7271348fb)

**Comment by Sqaaakoi** [comment]
nvm i'm wrong. This is a stock Discord issue that we can fix (and need to fix). Discord is using `MemberSafetyStore.getEnhancedMember(guildId, userId)` to get the highest role, and opening Mod View from the context menu does not try to load that information. See attached annotated screenshot for what needs to be changed to fix this.

![Diagram showing to remove MemberSafetyStore.getEnhancedMember and related get role call and to just find the highest role from the member's (passed to the component) role list and look it up in the same object like the listing of all other roles](https://github.com/user-attachments/assets/1046b7e8-9c9b-4d75-8aae-87ebee908ccc)

**Comment by Sqaaakoi** [comment]
tl;dr: discord overcomplicated things and depends on stuff that isnt important at all here
